packages/trader/src/App/Components/Elements/ContractDrawer/contract-drawer.jsx

```diff
@@ -24,13 +24,28 @@ import {
     getIndicativePrice,
     getEndTime,
     isUserSold,
-    isValidToSell       }      from 'Stores/Modules/Contract/Helpers/logic';
+    isValidToSell }            from 'Stores/Modules/Contract/Helpers/logic';
+import { memoize }             from 'Utils/React/memoize';
 import ContractCardBody        from './contract-card-body.jsx';
 import ContractCardFooter      from './contract-card-footer.jsx';
 import ContractCardHeader      from './contract-card-header.jsx';
 import ContractCard            from './contract-card.jsx';
 import Money                   from '../money.jsx';
 
+const memoizedIsUserSold = memoize(isUserSold);
+const memoizedIsValidToSell = memoize(isValidToSell);
+const memoizedGetEndTime = memoize(getEndTime);
+const memoizedGetIndicativePrice = memoize(getIndicativePrice);
+const memoizedGetCurrentTick = memoize(getCurrentTick);
+const memoizedGetDurationPeriod = memoize(getDurationPeriod);
+const memoizedGetDurationTime = memoize(getDurationTime);
+const memoizedGetDurationUnitText = memoize(getDurationUnitText);
```

Comment:
Hmm do we really want to memoize all these? e.g. for `getIndicativePrice` the output can be different even though the input is the same I think.

Reply:
```
export const getIndicativePrice = (contract_info) => (
    getFinalPrice(contract_info) && isEnded(contract_info) ?
        getFinalPrice(contract_info) :
        (+contract_info.bid_price || null)
);```

Since all of its calculation is based on `contract_info`. I don't think that statement is true. We are trading memory with CPU here, since our app is CPU heavy, I think this is a trade-off we can afford.

Reply:
oh yeah correct 👍